teams-notifier-core/src/main/java/fish/payara/extensions/notifiers/teams/TeamsNotifier.java

```diff
@@ -0,0 +1,154 @@
+ * enclosed by brackets [] replaced by your own identifying information:
+ * "Portions Copyright [year] [name of copyright owner]"
+ *
+ * Contributor(s):
+ * If you wish your version of this file to be governed by only the CDDL or
+ * only the GPL Version 2, indicate your decision by adding "[Contributor]
+ * elects to include this software in this distribution under the [CDDL or GPL
+ * Version 2] license."  If you don't indicate a single choice of license, a
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package fish.payara.extensions.notifiers.teams;
+
+import org.jvnet.hk2.annotations.Service;
+
+import fish.payara.internal.notification.PayaraConfiguredNotifier;
+import fish.payara.internal.notification.PayaraNotification;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.ConnectException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.ProtocolException;
+import java.net.URL;
+import java.net.UnknownHostException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import javax.json.Json;
+import javax.json.JsonObjectBuilder;
+import javax.ws.rs.HttpMethod;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+
+/**
+ * Notifier to integrate with Microsoft Teams
+ * @author jonathan coustick
+ */
+@Service(name = "teams-notifier")
+public class TeamsNotifier extends PayaraConfiguredNotifier<TeamsNotifierConfiguration> {
+    
+    private static final Logger LOGGER = Logger.getLogger(TeamsNotifier.class.getPackage().toString());
+ 
+    private static final String USER_AGENT = "Payara-Teams-Notifier";
+    
+    private URL url;
+    
+    @Override
+    public void handleNotification(PayaraNotification event) {
+        if (url == null) {
+            LOGGER.fine("Teams notifier received notification, but no URL was available.");
+            return;
+        }
+
+        try {
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+            connection.setDoOutput(true);
+            connection.setRequestMethod(HttpMethod.POST);
+            connection.setRequestProperty(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON);
+            connection.addRequestProperty(HttpHeaders.USER_AGENT, USER_AGENT);
+            
+            
+            try (OutputStream outputStream = connection.getOutputStream()) {
+                outputStream.write(eventToCard(event));
+                outputStream.flush();
+            }
+            connection.connect();
```

Comment:
Shouldn't this be `disconnect` and `connect` be directly after the request properties?

Reply:
Don't think so. The guides I looked at used this pattern.

Reply:
If you're writing data without connecting, it's probably working more by luck than judgment. Even if it works, it doesn't hit me as best practice